sdk/scheduler/src/main/java/com/mesosphere/sdk/scheduler/plan/DeploymentStep.java

```diff
@@ -84,10 +96,13 @@ public void updateOfferStatus(Collection<OfferRecommendation> recommendations) {
         setTaskIds(recommendations);
 
         if (recommendations.isEmpty()) {
-            setStatus(Status.PREPARED);
+            tasks.keySet().forEach(id -> setTaskStatus(id, Status.PREPARED));
         } else {
-            setStatus(Status.STARTING);
+            getTaskIds(recommendations).forEach(id -> setTaskStatus(id, Status.STARTING));
         }
+
+        prepared.set(true);
```

Comment:
setting `prepared` here before reading `prepared` in `getStatus(tasks)` looks risky.